Xamarin.Forms.Core/BindableProxy.cs

```diff
@@ -0,0 +1,336 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Reflection;
+
+namespace Xamarin.Forms
+{
+	class BindableProxy : BindableObject, IDisposable
+	{
+		readonly object _targetObject;
+		readonly string _targetProperty;
+		readonly string _targetEventName;
+		readonly PropertyInfo _propInfo;
+		List<MethodInfo> _setMethodsInfo;
+		List<MethodInfo> _getMethodsInfo;
+		List<Type> _possibleParameterTypes;
+		INativeValueConverter nativeValueConverter;
+		Action<object, object> _callbackSetValue;
+		Func<object> _callbackGetValue;
+		bool _isSettingFromProxy;
+
+		NativeViewEventListener _eventListener;
+		bool _disposed;
+
+
+		public BindableProxy(object target, PropertyInfo targetPropInfo, string eventName = null)
+		{
+			if (target == null)
+				throw new ArgumentNullException(nameof(target));
+			if (targetPropInfo == null)
+				throw new ArgumentException("targetProperty should not be null or empty", nameof(targetPropInfo));
+			_targetObject = target;
+			_targetProperty = targetPropInfo.Name;
+			_targetEventName = eventName;
+
+			_propInfo = targetPropInfo;
+
+			Init();
+		}
+
+		public BindableProxy(object target, string targetProp, Action<object, object> setValue = null, Func<object> getValue = null, string eventName = null)
```

Comment:
There's no need to have a BindableProxy per targetProperty. One might want to bind multiple properties to the same object